src/context.rs

```diff
@@ -940,6 +949,30 @@ mod tests {
         assert_eq!(expected_logical_dir, &context.logical_dir);
     }
 
+    #[test]
+    fn set_config_method_overwrites_constructor() {
+        let test_path = Path::new("/test_path").to_path_buf();
+        let context = Context::new_with_shell_and_path(
```

Comment:
would it be better to use `default_context()` instead? i.e.
```rust
let context = default_context();
let mod_context = default_context().set_config(...
```

Reply:
Sounds good! Let me do that now~